File: spring-lazyconnectiondatasourceproxy/src/main/java/com/study/lazyconnectiondatasourceproxy/domain/UserEntity.java
package com.study.lazyconnectiondatasourceproxy.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "users")
public class UserEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private int age;

    public UserEntity(String name, int age) {
        this.name = name;
        this.age = age;
    }
}

File: kotest/src/main/kotlin/io/devchw/kotest/infra/storage/core/jpa/user/UserEntity.kt
package io.devchw.kotest.infra.storage.core.jpa.user

import io.devchw.kotest.domain.user.User
import io.devchw.kotest.infra.storage.core.jpa.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Entity
@Table(name = "user")
class UserEntity(
    @Column(name = "name")
    val name: String,

    @Column(name = "age")
    val age: Int,

    @Column(name = "email")
    val email: String,
) : BaseEntity() {

    fun toDomain(): User {
        return User(
            id = id,
            name = name,
            age = age,
            email = email,
            createdAt = createdAt,
            updatedAt = updatedAt,
        )
    }

    companion object {
        fun of(user: User): UserEntity {
            return UserEntity(
                name = user.name,
                age = user.age,
                email = user.email,
            )
        }
    }
}